packages/contracts-bedrock/scripts/L2Genesis.s.sol

```diff
@@ -0,0 +1,432 @@
+        vm.etch(Predeploys.GOVERNANCE_TOKEN, address(token).code);
+
+        bytes32 _nameSlot = hex"0000000000000000000000000000000000000000000000000000000000000003";
+        bytes32 _symbolSlot = hex"0000000000000000000000000000000000000000000000000000000000000004";
+        bytes32 _ownerSlot = hex"000000000000000000000000000000000000000000000000000000000000000a";
+
+        vm.store(Predeploys.GOVERNANCE_TOKEN, _nameSlot, vm.load(address(token), _nameSlot));
+        vm.store(Predeploys.GOVERNANCE_TOKEN, _symbolSlot, vm.load(address(token), _symbolSlot));
+        vm.store(Predeploys.GOVERNANCE_TOKEN, _ownerSlot, bytes32(uint256(uint160(cfg.governanceTokenOwner()))));
+
+        /// Reset so its not included state dump
+        vm.etch(address(token), "");
+        vm.resetNonce(address(token));
+    }
+
+    /// @notice This predeploy is following the saftey invariant #1.
+    ///         This contract has no initializer.
+    /// @dev Previously the initial L1 attributes was set at genesis, to simplify,
+    ///      they no longer are so the resulting storage slots are no longer set.
+    function _setL1Block() internal {
+        _setImplementationCode(Predeploys.L1_BLOCK_ATTRIBUTES, "L1Block");
+    }
+
+    /// @dev Returns true if the address is not proxied.
+    function _notProxied(address _addr) internal pure returns (bool) {
+        return _addr == Predeploys.GOVERNANCE_TOKEN || _addr == Predeploys.WETH9;
+    }
+
+    /// @dev Returns true if the address is a predeploy.
+    function _isDefinedPredeploy(address _addr) internal pure returns (bool) {
+        return _addr == Predeploys.L2_TO_L1_MESSAGE_PASSER || _addr == Predeploys.L2_CROSS_DOMAIN_MESSENGER
+            || _addr == Predeploys.L2_STANDARD_BRIDGE || _addr == Predeploys.L2_ERC721_BRIDGE
+            || _addr == Predeploys.SEQUENCER_FEE_WALLET || _addr == Predeploys.OPTIMISM_MINTABLE_ERC20_FACTORY
+            || _addr == Predeploys.OPTIMISM_MINTABLE_ERC721_FACTORY || _addr == Predeploys.L1_BLOCK_ATTRIBUTES
+            || _addr == Predeploys.GAS_PRICE_ORACLE || _addr == Predeploys.DEPLOYER_WHITELIST || _addr == Predeploys.WETH9
+            || _addr == Predeploys.L1_BLOCK_NUMBER || _addr == Predeploys.LEGACY_MESSAGE_PASSER
+            || _addr == Predeploys.PROXY_ADMIN || _addr == Predeploys.BASE_FEE_VAULT || _addr == Predeploys.L1_FEE_VAULT
+            || _addr == Predeploys.GOVERNANCE_TOKEN || _addr == Predeploys.SCHEMA_REGISTRY || _addr == Predeploys.EAS;
+    }
+
+    /// @dev Function to compute the expected address of the predeploy implementation
+    ///      in the genesis state.
+    function _predeployToCodeNamespace(address _addr) internal pure returns (address) {
+        return address(
+            uint160(uint256(uint160(_addr)) & 0xffff | uint256(uint160(0xc0D3C0d3C0d3C0D3c0d3C0d3c0D3C0d3c0d30000)))
+        );
+    }
+
+    function _setImplementationCode(address _predeployAddr, string memory _predeployName) internal returns (address) {
```

Comment:
slight style nit: `_addr` and `_name` then use natspec to describe them more. Just a personal preference you dont have to follow this but it feels a lot cleaner to me